src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -214,59 +213,72 @@ export default class AdyenV2PaymentStrategy implements PaymentStrategy {
         });
     }
 
-    private _mountComponent(paymentMethodName: string): void {
-        const adyenv2 = this._getAdyenV2PaymentInitializeOptions();
-        const adyenCheckout = this._getAdyenCheckout();
-
-        switch (paymentMethodName) {
-            case AdyenPaymentMethodType.CreditCard:
-            case AdyenPaymentMethodType.ACH:
-            case AdyenPaymentMethodType.Bancontact:
-            case AdyenPaymentMethodType.GiroPay:
-            case AdyenPaymentMethodType.iDEAL:
-            case AdyenPaymentMethodType.SEPA:
-                const paymentComponent = adyenCheckout.create(paymentMethodName, {
-                        ...adyenv2.options,
-                        onChange: componentState => this._updateComponentState(componentState),
-                    }
-                );
+    private _mountCardVerificationComponent(): Promise<AdyenComponent> {
+        const adyenv2 = this._getPaymentInitializeOptions();
+        const adyenClient = this._getAdyenClient();
+        let cardVerificationComponent: AdyenComponent;
+
+        return new Promise(resolve => {
+            if (adyenv2.cardVerificationContainerId) {
+                cardVerificationComponent = adyenClient.create(AdyenComponentType.SecuredFields, {
+                    ...adyenv2.options,
+                    onChange: componentState => this._updateComponentState(componentState),
+                    onError: componentState => this._updateComponentState(componentState),
+                });
 
-                paymentComponent.mount(`#${adyenv2.containerId}`);
+                cardVerificationComponent.mount(`#${adyenv2.cardVerificationContainerId}`);
```

Comment:
what if this fails because the provided containerID doesnt exist? shouldn't you need to reject the returned promise?